bundler/lib/bundler/source_list.rb

```diff
@@ -37,6 +37,10 @@ def aggregate_global_source?
       global_rubygems_source.multiple_remotes?
     end
 
+    def implicit_global_source?
+      !global_rubygems_source.no_remotes?
```

Comment:
As pointed out in other review comment, I think the implementation here is inversed. It should be:

```suggestion
      global_rubygems_source.no_remotes?
```